packages/react-dom/src/client/ReactDOM.js

```diff
@@ -245,4 +251,192 @@ if (__DEV__) {
   }
 }
 
+if (__TEST__) {
+  const {IsSomeRendererActing} = ReactSharedInternals;
```

Comment:
Let's move this into the reconciler? The testing builds don't need to rely on shared state, except for the "overlapping act calls" warning.

This would also allow us to delete the ReactNoop and ReactTestRenderer copies of `act`, because they can just use one exported by the reconciler.